src/main/java/com/minecolonies/coremod/colony/workorders/WorkOrderBuild.java

```diff
@@ -296,98 +169,40 @@ private void sendBuilderMessage(@NotNull final Colony colony, final boolean hasB
         }
     }
 
-    private static boolean isLocationTownhall(@NotNull final Colony colony, final BlockPos buildingLocation)
-    {
-        return colony.hasTownHall() && colony.getTownHall() != null && colony.getTownHall().getID().equals(buildingLocation);
-    }
-
-    /**
-     * Returns the level up level of the building.
-     *
-     * @return Level after upgrade.
-     */
-    public int getUpgradeLevel()
-    {
-        return upgradeLevel;
-    }
-
-    /**
-     * Returns the ID of the building (aka ChunkCoordinates).
-     *
-     * @return ID of the building.
-     */
-    public BlockPos getBuildingLocation()
-    {
-        return buildingLocation;
-    }
-
-    /**
-     * Get the name the structure for this work order.
-     *
-     * @return the internal string for this structure.
-     */
-    public String getStructureName()
-    {
-        return structureName;
-    }
-
-    /**
-     * Gets how many times this structure should be rotated.
-     *
-     * @return building rotation.
-     */
-    public int getRotation()
+    @Override
+    public int getRotation(final World world)
     {
+        if (buildingRotation == 0 && world != null)
+        {
+            final IBlockState blockState = world.getBlockState(buildingLocation);
+            if (blockState.getBlock() instanceof AbstractBlockHut)
+            {
+                return BlockUtils.getRotationFromFacing(blockState.getValue(AbstractBlockHut.FACING));
+            }
+        }
         return buildingRotation;
     }
 
     /**
-     * Gets whether or not the building has been cleared.
-     *
-     * @return true if the building has been cleared.
-     */
-    public boolean isCleared()
-    {
-        return cleared;
-    }
-
-    /**
-     * Set whether or not the building has been cleared.
-     *
-     * @param cleared true if the building has been cleared.
-     */
-    public void setCleared(final boolean cleared)
-    {
-        this.cleared = cleared;
-    }
-
-    /**
-     * Gets whether or not the building materials have been requested already.
-     *
-     * @return true if the materials has been requested.
-     */
-    public boolean isRequested()
-    {
-        return requested;
-    }
-
-    /**
-     * Set whether or not the building materials have been requested already.
+     * Executed when a work order is completed.
      *
-     * @param requested true if so.
+     * Override this when something need to be done when the work order is completed
      */
-    public void setRequested(final boolean requested)
+    public void onCompleted(final Colony colony)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the "@Override" annotation above this method signature [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1161)